django/utils/html.py

```diff
@@ -153,11 +155,27 @@ def format_html_join(sep, format_string, args_generator):
       format_html_join('\n', "<li>{} {}</li>", ((u.first_name, u.last_name)
                                                   for u in users))
     """
-    return mark_safe(
-        conditional_escape(sep).join(
-            format_html(format_string, *args) for args in args_generator
+    if not isinstance(args_generator, Iterable):
+        raise TypeError("args_generator must be iterable.")
+
```

Comment:
We don’t need to add a type check. Any crash should be debuggable enough, runtime type checking adds unnecessary overhead, and type checkers exist to do type checking (though django-stubs is needed, at current...).
```suggestion
```